hack/verify-go-mod.sh

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 The Kubeflow Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+cd "$(dirname "$0")/.."
+
+GO_VERSION="$(grep '^go' go.mod | cut -d ' ' -f 2)"
+go mod tidy -go "${GO_VERSION}"
+git diff --exit-code -- 'go.*' > /dev/null || \
+  (echo "Please run \"go mod tidy -go ${GO_VERSION}\" to sync Go modules" && exit 1)
```

Comment:
Should we keep code generation check in 1 hack script ?
Then, we can reduce our number of scripts.

Reply:
Sure, thanks!